### ISSUE TITLE
Tpetra+coupled overset+periodic BC cause segfault in parallel

### ISSUE THREAD

**Issue by ashesh2512** [Exawind/nalu-wind#672]
## Detailed description of the bug 

This is a bug I encountered a couple of year ago, and encountered it again recently when running a simulation involving tptera, periodic bc, and coupled overset. 

After debugging, it was determined that the segfault occurs when `lid_b = 0` here - https://github.com/Exawind/nalu-wind/blob/master/src/TpetraLinearSystem.C#L759. This results in a negative array index here - https://github.com/Exawind/nalu-wind/blob/master/src/TpetraLinearSystemHelpers.C#L158. This seems to happen strictly for every node that is a donor for - the receptor/fringe nodes belonging to the mesh with the periodic boundaries. I also observed that the nodes in questions are being marked as ghost (probably because they exist on a separate MPI rank ?). I haven't been able to dig deeper. 

Note - I consider this to be a low priority bug because production runs won't involve periodic bc and coupled overset. The fact that no one else has encountered this in a couple of years is also telling about how rare this problem setup is. 

### Details to reproduce bug/build error

This issue still exists as of the SHA ids - 
`Nalu-Wind GIT Commit SHA: 28ada253f481cfe83b0446cb5fda4b07485a6db5`
`Trilinos Version: 12.17-g92ccd2b296`

I am attaching the input file and the exodus mesh - the problem is that of the decaying convecting taylor vortex from the verification manual - https://nalu-wind.readthedocs.io/en/latest/source/verification/verification.html#d-unsteady-uniform-property-convecting-decaying-taylor-vortex

[ctv.zip](https://github.com/Exawind/nalu-wind/files/5037558/ctv.zip)

**Comment by sayerhs** [comment]
See also NaluCFD/Nalu#89 and NaluCFD/Nalu#211
